tests/testthat/test-path.R

```diff
@@ -95,3 +95,16 @@ test_that("can write to a tar.gz file if the archive package is available", {
   expect_equal(res$path, "mtcars")
   expect_equal(res$size, 1281)
 })
+
+test_that("can read filepaths with non-ASCII characters", {
+  skip_on_cran()
+  ## chosen to be non-ASCII but
+  ## [1] representable in Windows-1252 and
+  ## [2] not any of the few differences between Windows-1252 and ISO-8859-1
+  ## a-grave + e-diaeresis  + Eth + '.csv'
+  tricky_filename <- "\u00C0\u00CB\u00D0\u002E\u0063\u0073\u0076"
+  file.copy(vroom_example("mtcars.csv"), tricky_filename)
+  expect_true(file.exists(tricky_filename))
+  on.exit(file.remove(tricky_filename))
```

Comment:
On principle, even though it does not matter here, I would add `add = TRUE` to the `on.exit()` call and place it right after the file creation code. This is always safest. The `add = TRUE` is important because it really should be the default for `on.exit()`: usually you want to add stuff to do later, not replace all previously scheduled todo's (note that readxl's tests currently have some mistakes in this regard!). The placement-right-after-creation is to minimize the chance that something else goes wrong, before the `on.exit()` call, resulting in us failing to clean up after ourselves. All of this is general advice about scheduling actions to happen later.

But here in this test, I'd actually prefer that you use `withr::with_tempfile()` or, probably, `withr::local_tempfile()`. In general, I'd prefer that we convert vroom's tests to use withr, whenever we touch them.

https://withr.r-lib.org/reference/with_tempfile.html

These are both nitpicky things. I'm just using this opportunity to discuss this topics, which will come up often in the future.